client/src/utils/helpers.js

```diff
@@ -15,14 +15,16 @@ var _ = {
     }
   },
   eachAsync: function(obj, fn, cb){
-    var c = obj.length;
+    var callbacksLeft = obj.length;
     
-    var next = function(){
-      c--;
-      if(c === 0){ 
-        var a = cb;
+    var next = function(err){
+      callbacksLeft--;
+      if(callbacksLeft === 0 || !!err){
+
+        var cbCopy = cb;
```

Comment:
I do not quite understand why you do cbCopy (not entirely sure you do copy it btw) and also you do _.noop ... can you explain as I am really curious about it?